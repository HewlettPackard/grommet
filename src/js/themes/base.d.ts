import { DeepReadonly, ColorType, OpacityType, BackgroundType } from '../utils'

export declare const base: DeepReadonly<ThemeType>;
export declare const generate: (baseSpacing: number, scale: number) => DeepReadonly<ThemeType>;

declare const colors: {
  active: ColorType;
  black: ColorType;
  border: ColorType;
  brand: ColorType;
  control: ColorType;
  focus: ColorType;
  placeholder: ColorType;
  selected: ColorType;
  text: ColorType;
  icon: ColorType;
  white: ColorType;
};
  
type Colors = typeof colors & {
  'accent-1': string;
  'accent-2': string;
  'accent-3': string;
  'accent-4': string;
  'neutral-1': string;
  'neutral-2': string;
  'neutral-3': string;
  'neutral-4': string;
  'dark-1': string;
  'dark-2': string;
  'dark-3': string;
  'dark-4': string;
  'dark-5': string;
  'dark-6': string;
  'light-1': string;
  'light-2': string;
  'light-3': string;
  'light-4': string;
  'light-5': string;
  'light-6': string;
  'status-critical': string;
  'status-error': string;
  'status-warning': string;
  'status-ok': string;
  'status-unknown': string;
  'status-disabled': string;
};

export interface ThemeType {
  global: {
    animation: {
      duration: string;
      jiggle: {
        duration: string;
      };
    };
    borderSize: {
      xsmall: string;
      small: string;
      medium: string;
      large: string;
      xlarge: string;
    };
    breakpoints: {
      small: {
        value: number;
        borderSize: {
          xsmall: string;
          small: string;
          medium: string;
          large: string;
          xlarge: string;
        };
        edgeSize: {
          none: string;
          hair: string;
          xxsmall: string;
          xsmall: string;
          small: string;
          medium: string;
          large: string;
          xlarge: string;
        };
        size: {
          xxsmall: string;
          xsmall: string;
          small: string;
          medium: string;
          large: string;
          xlarge: string;
          full: string;
        };
      };
      medium: {
        value: number;
      };
      large: {};
    };
    deviceBreakpoints: {
      phone: string;
      tablet: string;
      computer: string;
    };
    colors: Colors;
    control: {
      border: {
        width: string;
        radius: string;
        color: ColorType;
      };
    };
    debounceDelay: number;
    drop: {
      background: BackgroundType;
      border: {
        width: string;
        radius: string;
      };
      shadowSize: string;
      zIndex: string;
    };
    edgeSize: {
      none: string;
      hair: string;
      xxsmall: string;
      xsmall: string;
      small: string;
      medium: string;
      large: string;
      xlarge: string;
      responsiveBreakpoint: string;
    };
    elevation: {
      light: {
        none: string;
        xsmall: string;
        small: string;
        medium: string;
        large: string;
        xlarge: string;
      };
      dark: {
        none: string;
        xsmall: string;
        small: string;
        medium: string;
        large: string;
        xlarge: string;
      };
    };
    focus: {
      border: {
        color: ColorType;
      };
    };
    font: {
      size: string;
      height: string;
      maxWidth: string;
    };
    hover: {
      background: {
        dark: {
          color: ColorType;
          opacity: OpacityType;
        };
        light: {
          color: ColorType;
          opacity: OpacityType;
        };
      };
      color: ColorType;
    };
    input: {
      padding: string;
      weight: number;
    };
    opacity: {
      strong: number;
      medium: number;
      weak: number;
    };
    selected: {
      background: string;
      color: ColorType;
    };
    spacing: string;
    size: {
      xxsmall: string;
      xsmall: string;
      small: string;
      medium: string;
      large: string;
      xlarge: string;
      xxlarge: string;
      full: string;
    };
  };
  accordion: {
    icons: {
      collapse: any;
      expand: any;
    };
  };
  anchor: {
    textDecoration: string;
    fontWeight: number;
    color: ColorType;
    hover: {
      textDecoration: string;
    };
  };
  box: {
    responsiveBreakpoint: string;
  };
  button: {
    border: {
      width: string;
      radius: string;
    };
    primary: {};
    disabled: {
      opacity: number;
    };
    minWidth: string;
    maxWidth: string;
    padding: {
      vertical: string;
      horizontal: string;
    };
  };
  calendar: {
    small: {
      fontSize: string;
      lineHeight: number;
      daySize: string;
      slideDuration: string;
    };
    medium: {
      fontSize: string;
      lineHeight: number;
      daySize: string;
      slideDuration: string;
    };
    large: {
      fontSize: string;
      lineHeight: number;
      daySize: string;
      slideDuration: string;
    };
    icons: {
      previous: any;
      next: any;
      small: {
        previous: any;
        next: any;
      };
    };
  };
  carousel: {
    icons: {
      current: any;
      next: any;
      previous: any;
    };
  };
  checkBox: {
    border: {
      color: ColorType;
      width: string;
    };
    check: {
      extend: string;
      radius: string;
      thickness: string;
    };
    extend: string;
    color: ColorType;
    icon: { size: string; extend: string };
    icons: {};
    hover: {
      border: {
        color: ColorType;
      };
    };
    size: string;
    toggle: {
      color: ColorType;
      background: {
        dark: string;
        light: string;
      };
      radius: string;
      size: string;
      knob: { extend: string };
    };
  };
  clock: {
    analog: {
      hour: {
        color: ColorType;
        width: string;
        size: string;
        shape: string;
      };
      minute: {
        color: ColorType;
        width: string;
        size: string;
        shape: string;
      };
      second: {
        color: ColorType;
        width: string;
        size: string;
        shape: string;
      };
      size: {
        small: string;
        medium: string;
        large: string;
        xlarge: string;
        huge: string;
      };
    };
    digital: {
      text: {
        xsmall: {
          size: string;
          height: number;
        };
        small: {
          size: string;
          height: number;
        };
        medium: {
          size: string;
          height: number;
        };
        large: {
          size: string;
          height: number;
        };
        xlarge: {
          size: string;
          height: number;
        };
        xxlarge: {
          size: string;
          height: number;
        };
      };
    };
  };
  collapsible: {
    minSpeed: number;
    baseline: number;
  };
  dataTable: {
    header: {};
    groupHeader: {
      border: {
        side: string;
        size: string;
      };
      fill: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
      background: {
        dark: string;
        light: string;
      };
    };
    icons: {
      ascending: any;
      contract: any;
      descending: any;
      expand: any;
    };
    resize: {
      border: {
        side: string;
        color: ColorType;
      };
    };
    primary: {
      weight: string;
    };
  };
  formField: {
    border: {
      color: ColorType;
      position: string;
      side: string;
      error: {
        color: ColorType;
      };
    };
    error: {
      color: ColorType;
    };
    help: {
      color: ColorType;
    };
    label: {};
  };
  grommet: {};
  heading: {
    font: {};
    level: {
      1: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      2: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      3: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      4: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      5: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      6: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
    };
    responsiveBreakpoint: string;
    weight: number;
  };
  icon: {
    size: {
      small: string;
      medium: string;
      large: string;
      xlarge: string;
    };
  };
  layer: {
    background: string;
    border: {
      radius: string;
    };
    container: {
      zIndex: string;
    };
    overlay: {
      background: string;
    };
    responsiveBreakpoint: string;
    zIndex: string;
  };
  menu: {
    icons: {
      down: any;
    };
  };
  paragraph: {
    small: {
      size: string;
      height: string;
      maxWidth: string;
    };
    medium: {
      size: string;
      height: string;
      maxWidth: string;
    };
    large: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xxlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
  };
  radioButton: {
    border: {
      color: ColorType;
      width: string;
    };
    check: {
      radius: string;
    };
    hover: {
      border: {
        color: ColorType;
      };
    };
    icon: {};
    icons: {};
    gap: string;
    size: string;
  };
  rangeInput: {
    track: {
      height: string;
      color: any;
    };
    thumb: {};
  };
  select: {
    container: {};
    control: {};
    icons: {
      down: any;
    };
    step: number;
  };
  tab: {
    active: {
      color: ColorType;
    };
    border: {
      side: string;
      size: string;
      color: ColorType;
      active: {
        color: ColorType;
      };
      hover: {
        color: ColorType;
      };
    };
    color: ColorType;
    hover: {
      color: ColorType;
    };
    margin: {
      vertical: string;
      horizontal: string;
    };
    pad: {
      bottom: string;
    };
  };
  tabs: {
    header: {};
    panel: {};
  };
  table: {
    header: {
      align: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
      border: string;
      verticalAlign: string;
      fill: string;
    };
    body: {
      align: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
    };
    footer: {
      align: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
      border: string;
      verticalAlign: string;
      fill: string;
    };
  };
  text: {
    xsmall: {
      size: string;
      height: string;
      maxWidth: string;
    };
    small: {
      size: string;
      height: string;
      maxWidth: string;
    };
    medium: {
      size: string;
      height: string;
      maxWidth: string;
    };
    large: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xxlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
  };
  video: {
    captions: {
      background: string;
    };
    icons: {
      closedCaption: any;
      configure: any;
      fullScreen: any;
      pause: any;
      play: any;
      reduceVolume: any;
      volume: any;
    };
  };
  worldMap: {
    color: ColorType;
    continent: {
      active: string;
      base: string;
    };
    hover: {
      color: ColorType;
    };
    place: {
      active: string;
      base: string;
    };
  };
}
