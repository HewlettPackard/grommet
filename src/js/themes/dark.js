import { rgba } from 'polished';
import { css } from 'styled-components';

import { deepFreeze } from '../utils/object';
import { normalizeColor } from '../utils/colors';

const controlColor = '#FFCA58';
const accentColors = ['#FD6FFF', '#60EB9F', '#60EBE1', '#FFCA58'];
const neutralColors = ['#EB6060', '#01C781', '#6095EB', '#FFB200'];
const statusColors = {
  critical: '#FF3333',
  error: '#FF3333',
  warning: '#F7E464',
  ok: '#7DD892',
  unknown: '#a8a8a8',
  disabled: '#a8a8a8',
};
const backgroundColor = '#111111';
const textColor = '#eeeeee';
const borderColor = rgba(255, 255, 255, 0.33);
const activeColor = rgba(102, 102, 102, 0.5);

const colors = {
  active: activeColor,
  background: backgroundColor,
  black: '#000000',
  border: borderColor,
  brand: '#FD6FFF',
  control: controlColor,
  focus: controlColor,
  placeholder: '#AAAAAA',
  text: textColor,
  white: '#FFFFFF',
};

const colorArray = (array, prefix) =>
  array.forEach((color, index) => {
    colors[`${prefix}-${index + 1}`] = color;
  });

colorArray(accentColors, 'accent');
colorArray(neutralColors, 'neutral');
Object.keys(statusColors).forEach(color => {
  colors[`status-${color}`] = statusColors[color];
});

export const dark = deepFreeze({
  global: {
    colors,
    drop: {
      background: '#333333',
    },
    focus: {
      border: {
        color: css`
          ${props => normalizeColor('focus', props.theme)};
        `,
        width: '2px',
      },
    },
    font: {
      family: 'Arial',
    },
    input: {
      weight: 700,
    },
    text: {
      dark: textColor,
      light: '#000000',
    },
  },
  anchor: {
    color: controlColor,
  },
  icon: {
    color: textColor,
    colors,
  },
  layer: {
    background: backgroundColor,
    overlay: {
      background: rgba(48, 48, 48, 0.5),
    },
  },
});
