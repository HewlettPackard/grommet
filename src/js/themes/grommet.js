import { css } from 'styled-components';

import { deepFreeze } from '../utils';

const workSansPath = 'https://fonts.gstatic.com/s/worksans/v2';
const zillaSlabPath = 'https://fonts.gstatic.com/s/zillaslab/v3';

export const grommet = deepFreeze({
  global: {
    font: {
      family: "'Work Sans', Arial, sans-serif",
      face: `
        @font-face {
          font-family: 'Work Sans';
          font-style: normal;
          font-weight: 400;
          src:
            local('Work Sans'),
            local('WorkSans-Regular'),
            url("${workSansPath}/ElUAY9q6T0Ayx4zWzW63VJBw1xU1rKptJj_0jans920.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Work Sans';
          font-style: normal;
          font-weight: 600;
          src:
            local('Work Sans SemiBold'),
            local('WorkSans-SemiBold'),
            url("${workSansPath}/z9rX03Xuz9ZNHTMg1_ghGRampu5_7CjHW5spxoeN3Vs.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Zilla Slab';
          font-style: normal;
          font-weight: 400;
          src:
            local('Zilla Slab Regular'),
            local('ZillaSlab-Regular'),
            url("${zillaSlabPath}/dFa6ZfeM_74wlPZtksIFajo6_V6LVlA.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Zilla Slab';
          font-style: normal;
          font-weight: 600;
          src:
            local('Zilla Slab SemiBold'),
            local('ZillaSlab-SemiBold'),
            url("${zillaSlabPath}/dFa5ZfeM_74wlPZtksIFYuUe6HOpW3pwfa0.woff2") format('woff2');
        }
      `,
    },
  },
  anchor: {
    textDecoration: 'underline',
    color: {
      dark: '#9060EB',
      light: '#9060EB',
    },
  },
  button: {
    extend: css`
      ${props => !props.plain && 'font-weight: bold;'}
    `,
  },
  checkBox: {
    icon: {
      extend: css`
        box-sizing: border-box;
        position: absolute;
        top: 0px;
        left: 0px;
        width: ${props => props.theme.checkBox.size};
        height: ${props => props.theme.checkBox.size};
      `,
    },
  },
  heading: {
    font: {
      family: "'Zilla Slab', 'Work Sans', Arial, sans-serif",
    },
  },
});
