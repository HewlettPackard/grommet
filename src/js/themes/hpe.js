import { deepFreeze } from '../utils';

const accentColors = ['#2AD2C9', '#614767', '#ff8d6d'];
const neutralColors = ['#425563', '#5F7A76', '#80746E', '#767676'];
const statusColors = {
  critical: '#F04953',
  error: '#F04953',
  warning: '#FFD144',
  ok: '#01a982',
  unknown: '#CCCCCC',
  disabled: '#CCCCCC',
};

const colors = {
  accent: accentColors,
  brand: '#01a982',
  focus: accentColors[0],
  neutral: neutralColors,
  status: statusColors,
};

const colorArray = (array, prefix) =>
  array.forEach((color, index) => {
    colors[`${prefix}-${index + 1}`] = color;
  });

colorArray(accentColors, 'accent');
colorArray(neutralColors, 'neutral');
Object.keys(statusColors).forEach((color) => {
  colors[`status-${color}`] = statusColors[color];
});

export default deepFreeze({
  global: {
    colors,
    font: {
      family: "'Metric', Arial, sans-serif",
      face: `
        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Regular.woff") format('woff');
        }

        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Bold.woff") format('woff');
          font-weight: 700;
        }

        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Semibold.woff") format('woff');
          font-weight: 600;
        }

        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Light.woff") format('woff');
          font-weight: 100;
        }
      `,
    },
  },
  anchor: {
    color: 'inherit',
  },
  button: {
    border: {
      radius: '0px',
    },
    colors: {
      accent: '#ff8d6d',
      secondary: 'rgba(51,51,51,0.6)',
    },
    extend: 'letter-spacing: 0.04167em;',
  },
  clock: {
    second: {
      color: {
        dark: '#01a982',
        light: '#01a982',
      },
    },
  },
  heading: {
    font: false,
  },
});
