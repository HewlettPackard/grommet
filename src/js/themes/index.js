import base from './base';

export { default as base } from './base';
export { default as grommet } from './grommet';
export { default as hpe } from './hpe';
export { default as dark } from './dark';

export default base;
