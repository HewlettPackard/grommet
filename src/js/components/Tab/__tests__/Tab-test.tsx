import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Grommet, Tabs } from '../..';
import { Tab } from '..';
import { ThemeType } from '../../../themes';

describe('Tab', () => {
  test('default', () => {
    const { container } = render(
      <Grommet>
        <Tabs>
          <Tab title="Title 1" />
        </Tabs>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should have no default styles with plain prop', () => {
    const { container, getByText } = render(
      <Grommet>
        <Tabs>
          <Tab title="Title 1" plain />
        </Tabs>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
    const plainTab = getByText('Title 1').parentElement!;
    const plainTabStyle = window.getComputedStyle(plainTab);
    expect(plainTabStyle.borderBottom).toBe('');
  });

  test('should allow to extend tab styles', () => {
    const customTheme: ThemeType = {
      tab: {
        extend: `color: red;
                padding: 20px;
                box-shadow: rgba(0, 0, 0, 0.25) 0px 54px 55px, rgba(0, 0, 0, 0.12) 0px -12px 30px, rgba(0, 0, 0, 0.12) 0px 4px 6px, rgba(0, 0, 0, 0.17) 0px 12px 13px, rgba(0, 0, 0, 0.09) 0px -3px 5px;
                margin: 30px;`,
      },
    };
    const { container, getByText } = render(
      <Grommet theme={customTheme}>
        <Tabs>
          <Tab title="Title 1" plain>
            Some content
          </Tab>
          <Tab title="Title 2">Some content 2</Tab>
        </Tabs>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
    const extendedPlainTab = getByText('Title 1')!;
    const extendedPlainTabStyle = window.getComputedStyle(extendedPlainTab);
    // color can be changed only when plain prop used
    expect(extendedPlainTabStyle.color).toBe('red');

    const extendedTab = getByText('Title 2')!;
    const extendedTabStyle = window.getComputedStyle(extendedTab);
    expect(extendedTabStyle.color).not.toBe('red');

    const extendedTabParent = extendedTab.parentElement!;
    const extendedTabParentStyle = window.getComputedStyle(extendedTabParent);
    expect(extendedTabParentStyle.padding).toBe('20px');
  });
});
