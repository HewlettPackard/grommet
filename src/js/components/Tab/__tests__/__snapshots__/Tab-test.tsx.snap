// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab default 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 3px;
  margin-bottom: 3px;
  border-bottom: solid 2px #000000;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 6px;
}

.c6 {
  font-size: 18px;
  line-height: 24px;
  color: #444444;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c3:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus > circle,
.c3:focus > ellipse,
.c3:focus > line,
.c3:focus > path,
.c3:focus > polygon,
.c3:focus > polyline,
.c3:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus::-moz-focus-inner {
  border: 0;
}

.c3:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible) > circle,
.c3:focus:not(:focus-visible) > ellipse,
.c3:focus:not(:focus-visible) > line,
.c3:focus:not(:focus-visible) > path,
.c3:focus:not(:focus-visible) > polygon,
.c3:focus:not(:focus-visible) > polyline,
.c3:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c5 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 3px;
  margin-bottom: 3px;
}

.c5:hover {
  color: #000000;
}

.c7 {
  min-height: 0;
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-top: 2px;
    margin-bottom: 2px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-bottom: solid 2px #000000;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding-bottom: 3px;
  }
}

<div
  class="c0"
>
  <div
    class="c1 "
    role="tablist"
  >
    <div
      class="c2 "
    >
      <button
        aria-expanded="true"
        aria-selected="true"
        class="c3"
        role="tab"
        type="button"
      >
        <div
          class="c4 c5"
        >
          <span
            class="c6"
          >
            Title 1
          </span>
        </div>
      </button>
    </div>
    <div
      aria-label="Title 1 Tab Contents"
      class="c7"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tab should allow to extend tab styles 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 3px;
  margin-bottom: 3px;
  border-bottom: solid 2px #7D4CDB;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 6px;
}

.c7 {
  font-size: 18px;
  line-height: 24px;
  color: #7D4CDB;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c3:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus > circle,
.c3:focus > ellipse,
.c3:focus > line,
.c3:focus > path,
.c3:focus > polygon,
.c3:focus > polyline,
.c3:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus::-moz-focus-inner {
  border: 0;
}

.c3:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible) > circle,
.c3:focus:not(:focus-visible) > ellipse,
.c3:focus:not(:focus-visible) > line,
.c3:focus:not(:focus-visible) > path,
.c3:focus:not(:focus-visible) > polygon,
.c3:focus:not(:focus-visible) > polyline,
.c3:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c4 {
  color: red;
  padding: 20px;
  box-shadow: rgba(0,0,0,0.25) 0px 54px 55px,rgba(0,0,0,0.12) 0px -12px 30px,rgba(0,0,0,0.12) 0px 4px 6px,rgba(0,0,0,0.17) 0px 12px 13px,rgba(0,0,0,0.09) 0px -3px 5px;
  margin: 30px;
}

.c6 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 3px;
  margin-bottom: 3px;
  color: red;
  padding: 20px;
  box-shadow: rgba(0,0,0,0.25) 0px 54px 55px,rgba(0,0,0,0.12) 0px -12px 30px,rgba(0,0,0,0.12) 0px 4px 6px,rgba(0,0,0,0.17) 0px 12px 13px,rgba(0,0,0,0.09) 0px -3px 5px;
  margin: 30px;
}

.c6:hover {
  color: #000000;
}

.c8 {
  min-height: 0;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-top: 2px;
    margin-bottom: 2px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border-bottom: solid 2px #7D4CDB;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding-bottom: 3px;
  }
}

<div
  class="c0"
>
  <div
    class="c1 "
    role="tablist"
  >
    <div
      class="c2 "
    >
      <button
        aria-expanded="true"
        aria-selected="true"
        class="c3"
        role="tab"
        type="button"
      >
        <div
          class="c1 c4"
        >
          Title 1
        </div>
      </button>
      <button
        aria-expanded="false"
        aria-selected="false"
        class="c3"
        role="tab"
        type="button"
      >
        <div
          class="c5 c6"
        >
          <span
            class="c7"
          >
            Title 2
          </span>
        </div>
      </button>
    </div>
    <div
      aria-label="Title 1 Tab Contents"
      class="c8"
      role="tabpanel"
    >
      Some content
    </div>
  </div>
</div>
`;

exports[`Tab should have no default styles with plain prop 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c3:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus > circle,
.c3:focus > ellipse,
.c3:focus > line,
.c3:focus > path,
.c3:focus > polygon,
.c3:focus > polyline,
.c3:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus::-moz-focus-inner {
  border: 0;
}

.c3:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible) > circle,
.c3:focus:not(:focus-visible) > ellipse,
.c3:focus:not(:focus-visible) > line,
.c3:focus:not(:focus-visible) > path,
.c3:focus:not(:focus-visible) > polygon,
.c3:focus:not(:focus-visible) > polyline,
.c3:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c4 {
  min-height: 0;
}

<div
  class="c0"
>
  <div
    class="c1 "
    role="tablist"
  >
    <div
      class="c2 "
    >
      <button
        aria-expanded="true"
        aria-selected="true"
        class="c3"
        role="tab"
        type="button"
      >
        <div
          class="c1 "
        >
          Title 1
        </div>
      </button>
    </div>
    <div
      aria-label="Title 1 Tab Contents"
      class="c4"
      role="tabpanel"
    />
  </div>
</div>
`;
